62.CAD/main.cpp

```diff
@@ -475,14 +577,66 @@ struct DrawBuffersFiller
 		return intendedNextSubmit;
 	}
 
-	// TODO[Lucas]: drawHatch function with similar signature to drawPolyline
 	// If we had infinite mem, we would first upload all curves into geometry buffer then upload the "CurveBoxes" with correct gpu addresses to those
 	// But we don't have that so we have to follow a similar auto submission as the "drawPolyline" function with some mutations:
 	// We have to find the MAX number of "CurveBoxes" we could draw, and since both the "Curves" and "CurveBoxes" reside in geometry buffer,
 	// it has to be taken into account when calculating "how many curve boxes we could draw and when we need to submit/clear"
 	// So same as drawPolylines, we would first try to fill the geometry buffer and index buffer that corresponds to "backfaces or even provoking vertices"
 	// then change index buffer to draw front faces of the curveBoxes that already reside in geometry buffer memory
 	// then if anything was left (the ones that weren't in memory for front face of the curveBoxes) we copy their geom to mem again and use frontface/oddProvoking vertex
+	video::IGPUQueue::SSubmitInfo drawHatch(
+		const Hatch& hatch,
+		// If more parameters from cpu line style are used here later, make a new HatchStyle & use that
+		const float32_t4 color, 
+		const uint32_t clipProjectionIdx,
+		video::IGPUQueue* submissionQueue,
+		video::IGPUFence* submissionFence,
+		video::IGPUQueue::SSubmitInfo intendedNextSubmit)
+	{
+		CPULineStyle lineStyle;
+		lineStyle.color = color;
+
+		uint32_t styleIdx;
+		intendedNextSubmit = addLineStyle_SubmitIfNeeded(lineStyle, styleIdx, submissionQueue, submissionFence, intendedNextSubmit);
+		
+		MainObject mainObj = {};
+		mainObj.styleIdx = styleIdx;
+		mainObj.clipProjectionIdx = clipProjectionIdx;
+		uint32_t mainObjIdx;
+		intendedNextSubmit = addMainObject_SubmitIfNeeded(mainObj, mainObjIdx, submissionQueue, submissionFence, intendedNextSubmit);
+
+		const auto sectionsCount = 1; //hatch.hatchBoxes.size();
```

Comment:
don't work with the vectors directly, make them private
add `Hatch::getHatchBoxesCount()` and `Hatch::getHatchBox(uint32_t idx)`, similar to how I work with quad beziers in CPolyline